app/models/attribute.rb

```diff
@@ -83,8 +85,29 @@ def as_txt(attributes, version:, locale: "en")
       ].join("\n")
     end
 
+    def sort_values(attribute_friendly_id, values, locale: "en")
+      if @custom_attribute_value_order_indexes&.dig(attribute_friendly_id)
+        return values.sort_by do |value|
+          @custom_attribute_value_order_indexes[attribute_friendly_id][value.friendly_id]
+        end
+      end
+
+      ValueSorter.sort_by_localized_name(values, locale: locale)
```

Comment:
I think the var extraction in the if ends up being easier to read? Unsure on the right var name for this one though 🤔 

```suggestion
      if (manual_sort = @custom_attribute_value_order_indexes&.dig(attribute_friendly_id))
        values.sort_by { manual_sort[_1.friendly_id] }
      else
        ValueSorter.sort_by_localized_name(values, locale: locale)
      end
```